### ISSUE TITLE
Error: Resource XXX must be in ARN format or "*" after amplify function update

### ISSUE THREAD

**Issue by amirhajimirsadeghi** [aws-amplify/amplify-cli#5333]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
In depth description of the bug with step by step commands and outputs provided under the screenshot section. 

Summary of bug is: I have a project with DDB Table, Lambda Function, and a Trigger from table to the lambda. I tried updating the function to have addition permissions to access dynamodb tables and then deployment fails because StreamArn no longer gets passed to the function CloudFormation as a parameter. 

**Amplify CLI Version**
4.29.1

**To Reproduce**
1. have environment with a DynamoDb table, lambda function,  and a trigger from the table to the function
2. Update function to have permission to access any table
3. amplify push

**Expected behavior**

I expected the function to update with new permissions but instead it failed because the Arn of the DynamoDB Stream was not passed in as a parameter.

**Screenshots**

1. From a working state I executed the following
```
➜  client-webapp git:(redesign) ✗ amplify function update                                                                                                                                
? Select which capability you want to update: Lambda function (serverless function)
? Select the Lambda function you want to update orderprocessing
? Do you want to update the Lambda function permissions to access other resources in this project? Yes
? Select the category storage
? Storage has 2 resources in this project. Select the one you would like your Lambda to access catalog, user
? Select the operations you want to permit for catalog create, read, update, delete
? Select the operations you want to permit for user create, read, update, delete

You can access the following resource attributes as environment variables from your Lambda function
	STORAGE_CATALOG_ARN
	STORAGE_CATALOG_NAME
	STORAGE_USER_ARN
	STORAGE_USER_NAME
? Do you want to invoke this function on a recurring schedule? No
? Do you want to configure Lambda layers for this function? No
? Do you want to edit the local lambda function now? No
```
```
amplify push
```
Then I get the following error:
```
UPDATE_FAILED               LambdaTriggerPolicy                                                        AWS::IAM::Policy           Tue Sep 15 2020 10:03:57 GMT-0500 (Central Daylight Time) Resource storageuserStreamArn must be in ARN format or "*". (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: 41bf7d8e-89fd-4393-81d2-1f1279887afe; Proxy: null)
UPDATE_FAILED               userTriggerPolicy                                                          AWS::IAM::Policy           Tue Sep 15 2020 10:03:57 GMT-0500 (Central Daylight Time) Resource storageuserStreamArn must be in ARN format or "*". (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: 8dbed913-12f9-4f39-a83a-9207c7a313f7; Proxy: null)
```

Here's my git diff from working version:
```diff
➜  client-webapp git:(redesign) ✗ git diff --exit-code                                                                                                                                   
diff --git a/amplify/backend/backend-config.json b/amplify/backend/backend-config.json
index 5ef61c4..47626fd 100644
--- a/amplify/backend/backend-config.json
+++ b/amplify/backend/backend-config.json
@@ -64,11 +64,21 @@
       "providerPlugin": "awscloudformation",
       "service": "Lambda",
       "dependsOn": [
+        {
+          "category": "storage",
+          "resourceName": "catalog",
+          "attributes": [
+            "Name",
+            "Arn"
+          ]
+        },
         {
           "category": "storage",
           "resourceName": "user",
           "attributes": [
-            "StreamArn"
+            "StreamArn",
+            "Name",
+            "Arn"
           ]
         }
       ]
diff --git a/amplify/backend/function/orderprocessing/function-parameters.json b/amplify/backend/function/orderprocessing/function-parameters.json
index d507877..efcf55e 100644
--- a/amplify/backend/function/orderprocessing/function-parameters.json
+++ b/amplify/backend/function/orderprocessing/function-parameters.json
@@ -1,3 +1,19 @@
 {
-  "lambdaLayers": []
+  "lambdaLayers": [],
+  "permissions": {
+    "storage": {
+      "catalog": [
+        "create",
+        "read",
+        "update",
+        "delete"
+      ],
+      "user": [
+        "create",
+        "read",
+        "update",
+        "delete"
+      ]
+    }
+  }
 }
\ No newline at end of file
diff --git a/amplify/backend/function/orderprocessing/orderprocessing-cloudformation-template.json b/amplify/backend/function/orderprocessing/orderprocessing-cloudformation-template.json
index e06bcdd..118bccf 100644
--- a/amplify/backend/function/orderprocessing/orderprocessing-cloudformation-template.json
+++ b/amplify/backend/function/orderprocessing/orderprocessing-cloudformation-template.json
@@ -13,6 +13,22 @@
     "storageuserStreamArn": {
       "Type": "String",
       "Default": "storageuserStreamArn"
+    },
+    "storagecatalogName": {
+      "Type": "String",
+      "Default": "storagecatalogName"
+    },
+    "storagecatalogArn": {
+      "Type": "String",
+      "Default": "storagecatalogArn"
+    },
+    "storageuserName": {
+      "Type": "String",
+      "Default": "storageuserName"
+    },
+    "storageuserArn": {
+      "Type": "String",
+      "Default": "storageuserArn"
     }
   },
   "Conditions": {
@@ -59,6 +75,18 @@
             },
             "REGION": {
               "Ref": "AWS::Region"
+            },
+            "STORAGE_CATALOG_NAME": {
+              "Ref": "storagecatalogName"
+            },
+            "STORAGE_CATALOG_ARN": {
+              "Ref": "storagecatalogArn"
+            },
+            "STORAGE_USER_NAME": {
+              "Ref": "storageuserName"
+            },
+            "STORAGE_USER_ARN": {
+              "Ref": "storageuserArn"
             }
           }
         },
@@ -263,6 +291,91 @@
         },
         "StartingPosition": "LATEST"
       }
+    },
+    "AmplifyResourcesPolicy": {
+      "DependsOn": [
+        "LambdaExecutionRole"
+      ],
+      "Type": "AWS::IAM::Policy",
+      "Properties": {
+        "PolicyName": "amplify-lambda-execution-policy",
+        "Roles": [
+          {
+            "Ref": "LambdaExecutionRole"
+          }
+        ],
+        "PolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Action": [
+                "dynamodb:Put*",
+                "dynamodb:Create*",
+                "dynamodb:BatchWriteItem",
+                "dynamodb:Get*",
+                "dynamodb:BatchGetItem",
+                "dynamodb:List*",
+                "dynamodb:Describe*",
+                "dynamodb:Scan",
+                "dynamodb:Query",
+                "dynamodb:Update*",
+                "dynamodb:RestoreTable*",
+                "dynamodb:Delete*"
+              ],
+              "Resource": [
+                {
+                  "Ref": "storagecatalogArn"
+                },
+                {
+                  "Fn::Join": [
+                    "/",
+                    [
+                      {
+                        "Ref": "storagecatalogArn"
+                      },
+                      "index/*"
+                    ]
+                  ]
+                }
+              ]
+            },
+            {
+              "Effect": "Allow",
+              "Action": [
+                "dynamodb:Put*",
+                "dynamodb:Create*",
+                "dynamodb:BatchWriteItem",
+                "dynamodb:Get*",
+                "dynamodb:BatchGetItem",
+                "dynamodb:List*",
+                "dynamodb:Describe*",
+                "dynamodb:Scan",
+                "dynamodb:Query",
+                "dynamodb:Update*",
+                "dynamodb:RestoreTable*",
+                "dynamodb:Delete*"
+              ],
+              "Resource": [
+                {
+                  "Ref": "storageuserArn"
+                },
+                {
+                  "Fn::Join": [
+                    "/",
+                    [
+                      {
+                        "Ref": "storageuserArn"
+                      },
+                      "index/*"
+                    ]
+                  ]
+                }
+              ]
+            }
+          ]
+        }
+      }
     }
   },
   "Outputs": {
```

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version: v14.7.0

**Additional context**

**Comment by amirhajimirsadeghi** [comment]
Pushing the changes and having it go through the pipeline works fine, but `amplify push` doesn't.

**Comment by montycheese** [comment]
I just got the same issue after doing a `amplify function update`

Were you able to resolve this issue?

**Comment by kenbankspeng** [comment]
Same issue here. Occurs after `amplify function update`.

**Comment by Leyko1** [comment]
I had the same issue. I figured that the **amplify-meta.json** was also updated and the **dependsOn** was overriden (this can also be the reason why `amplify push` is not working when the pipeline works fine)

Take a look at your **amplify-meta.json** and check if **StreamArn** is missing in **dependsOn**
```diff
...
"lambdaTriggerName": {
      "build": true,
      "providerPlugin": "awscloudformation",
      "service": "Lambda",
      "dependsOn": [
        {
          "category": "storage",
          "resourceName": "storageName",
          "attributes": [
            "Name",
            "Arn",
+           "StreamArn"
          ]
        }
      ],
...
```

_amplify-meta.json is in .gitignore by default._

**Comment by econtentmaps** [comment]
I am having this problem too. Adding the storage category as defined in @Leyko1 comment manually in amplify-meta.json fixed it locally, but on deploying via the Amplify console (checking in repo), the automated build fails with the same error. Does anyone have any idea how to fix that?

**Comment by econtentmaps** [comment]
Can you re-open this, please @UnleashedMind? I've had it happen again with the latest CLI.

**Comment by pr1ze** [comment]
Also happened for me after using "amplify function update".

Had a function which is triggered by kinesis data stream. Did an update to allow the function modify some new dynamodb tables i made. and now seeing the error.

Amplify cli verison 4.46.1

**Comment by tomoima525** [comment]
@econtentmaps  maybe you have figured it out, but you need to call `amplify env checkout {your environment}` to update amplify-meta.json

**Comment by kenbankspeng** [comment]
There seems to be a handful of inside secrets with amplify that are gained only through experience, and which of course can change with each new release. 

For instance, the things that always worked with a single user sometimes need special steps to work in multi-env, especially if the changes involve resources.

And I've seen some edge case changes that amplify was not able to pick up, like someone deleting and rebuilding a table with a slight variation under the same name, but amplify push doesn't actually see the change since same name.

**Comment by kevinlam92** [comment]
Here's a workaround that worked for me, and by no means this is a fix.

1.) I removed the trigger with `amplify storage update`
2.) In my trigger function's cloud formation template I hardcoded the trigger stream arn's default value to the dynamoDB stream arn.

The stream works and successfully calls my lambda function. However, this will break if the stream arn ever changes (ie. disable/re-enable), and you will have to reupdate it in the cloud formation template.

At least my `amplify push` works and unblocks me.
